app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -112,25 +112,62 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 <%= "          --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
 <%= "          --data 'packages=#{@packages}' \\\n" if @packages.present? -%>
 
-}
+    }
 
     <% if @force -%>
     yum remove -y katello-ca-consumer*
     <% end -%>
 
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
+    KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/katello-server-ca.crt
+    RHSM_CFG=/etc/rhsm/rhsm.conf
 
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
+    # Install or upgrade subscription-manager
+    if ! [ -x "$(command -v subscription-manager)" ] ; then
+      if [ -f /usr/bin/dnf ]; then
+        dnf install -y subscription-manager
+      else
+        yum install -y subscription-manager
+      fi
```

Comment:
In theory, we do also have sub-man on Debian (but not supported today by `bootrstrap.py`, and this line is guarded by "if RHEL or CentOS").

And I always think that the sheer presence of a binary should not mean it can be used blindly. My Debian systems have `/usr/bin/rpm` and my Fedora has `/usr/bin/dpkg` (those are not the binaries you're testing for, but you get the idea).

I think the correct approach should be to parse `/etc/os-release` and only if that fails fall back to "is that binary there". Probably good as a helper somewhere, as I expect this script to need "the right package manager" in a few places.